### PR TITLE
🌱 Testing: Add unit tests for NetworkPolicy controller, fetcher, webhook, proactive restart, and negative signature verification

### PR DESCRIPTION
**Pull request by kevincogan**
## Summary

- Add unit tests for the `AgentCardNetworkPolicyReconciler` covering enforcement toggle, permissive/restrictive policy creation, identity binding, policy status transitions, deletion cleanup, and missing workload handling
- Add unit tests for the `DefaultFetcher` covering A2A card fetch, unsupported protocol, HTTP error codes, invalid JSON, and `GetServiceURL` formatting
- Add unit tests for the `AgentCardValidator` webhook covering create/update/delete validation with and without `targetRef`, and wrong object type rejection
- Add integration tests for proactive workload restart on trust bundle rotation and SVID leaf cert expiry, including the no-restart negative case
- Add integration tests for negative signature verification covering tampered card payload rejection in enforce mode (1m requeue) and tampered card acceptance in audit mode

## Related issue(s)

Closes test coverage gaps identified in the AgentCard design review for the NetworkPolicy controller, fetcher, webhook, proactive restart, and signature verification paths.

### Fast unit tests (fetcher + webhook)
`go test ./internal/agentcard/... ./internal/webhook/... -v -count=1`

### Controller integration tests (requires envtest binaries)
`go test ./internal/controller/... -v -count=1 -timeout 120s`

### CODE REVIEW

File: kagenti-operator/internal/webhook/v1alpha1/agentcard_webhook_test.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
+package v1alpha1
```
Comment (pdettori) [inline kagenti-operator/internal/webhook/v1alpha1/agentcard_webhook_test.go]: suggestion: Missing Apache 2.0 license header. The other test files in this PR all have it — would be good to add for consistency.

File: kagenti-operator/internal/controller/negative_signature_test.go [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+var _ = Describe("Negative Signature Verification", func() {
+	Context("Tampered card payload in enforce mode", func() {
+		const (
+			deploymentName = "sig-tamper-agent"
+			agentCardName  = "sig-tamper-card"
+			namespace      = "default"
+		)
+
+		ctx := context.Background()
```
Comment (pdettori) [inline kagenti-operator/internal/controller/negative_signature_test.go]: suggestion: This shadows the suite-level `ctx` with `context.Background()`. The network policy controller tests use the suite-level `ctx` — would be more consistent to do the same here (and in `proactive_restart_test.go` lines 104, 162, 216). If the suite context carries cancellation or deadline semantics, these tests would miss them.

File: kagenti-operator/internal/controller/agentcard_networkpolicy_controller_test.go [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+			getPolicy(deploymentName, namespace)
+
+			toDelete := &agentv1alpha1.AgentCard{}
+			Expect(k8sClient.Get(ctx, types.NamespacedName{Name: agentCardName, Namespace: namespace}, toDelete)).To(Succeed())
+			Expect(k8sClient.Delete(ctx, toDelete)).To(Succeed())
+
+			reconcileNP(r, agentCardName, namespace)
+
+			Eventually(func() bool {
+				return k8sClient.Get(ctx, types.NamespacedName{Name: deploymentName + "-signature-policy", Namespace: namespace}, &netv1.NetworkPolicy{}) != nil
```
Comment (pdettori) [inline kagenti-operator/internal/controller/agentcard_networkpolicy_controller_test.go]: nit: This checks `!= nil` to verify the resource is gone, but that would also return true on a transient API error. Using `apierrors.IsNotFound(err)` would be more precise for deletion verification.

File: kagenti-operator/internal/controller/proactive_restart_test.go [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+	if err := k8sClient.Get(ctx, types.NamespacedName{Name: name, Namespace: ns}, d); err != nil {
+		return ""
+	}
+	if d.Spec.Template.Annotations == nil {
+		return ""
+	}
+	return d.Spec.Template.Annotations[AnnotationResignTrigger]
+}
+
+func reconcileTwice(reconciler *AgentCardReconciler, name, ns string) {
```
Comment (pdettori) [inline kagenti-operator/internal/controller/proactive_restart_test.go]: nit: A brief comment explaining *why* two reconciles are needed (e.g., first adds the finalizer, second does the actual work) would help future maintainers. The same double-reconcile pattern appears throughout the controller tests without explanation.

File: kagenti-operator/internal/controller/proactive_restart_test.go [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+			}
+			Expect(k8sClient.Create(ctx, agentCard)).To(Succeed())
+
+			reconciler := &AgentCardReconciler{
+				Client:            k8sClient,
+				Scheme:            k8sClient.Scheme(),
+				AgentFetcher:      &mockFetcherFunc{fn: makeCardData},
+				RequireSignature:  true,
+				SignatureProvider: &mockSignatureProviderWithBundleHash{
+					pubKeyPEM: pubPEM, bundleHash: "same-hash", leafExpiry: time.Now().Add(5 * time.Minute),
```
Comment (pdettori) [inline kagenti-operator/internal/controller/proactive_restart_test.go]: nit: The 5-minute grace period is a magic number here. If the production code defines a constant for the SVID grace period threshold, referencing it would make this test more resilient to future changes.

**Review by pdettori — APPROVED** [review]
## Review Summary

Solid set of tests (941 lines) covering critical paths: signature verification (positive/negative), network policy enforcement (permissive/restrictive/disabled), proactive restart logic, and webhook validation. Well-structured with appropriate frameworks (Ginkgo+envtest for controller tests, standard `testing.T` for pure functions) and good isolation with cleanup hooks and unique resource names.

**Areas reviewed**: Go tests, Security, Conventions
**Commits**: 1 commit, signed-off ✓
**CI status**: passing ✓

A few suggestions below for consistency and maintainability, but nothing blocking merge.
